test/e2e/helloworld_test.go

```diff
@@ -69,4 +70,37 @@ func TestHelloWorld(t *testing.T) {
 	if err != nil {
 		t.Fatalf("The endpoint for Route %s at domain %s didn't serve the expected text \"%s\": %v", names.Route, domain, helloWorldExpectedOutput, err)
 	}
+
+	var revName string
+	err = test.WaitForConfigurationState(clients.ServingClient, names.Config, func(c *v1alpha1.Configuration) (bool, error) {
+		if c.Status.LatestCreatedRevisionName != names.Revision {
+			revName = c.Status.LatestCreatedRevisionName
+			return true, nil
+		}
+		return false, nil
+	}, "ConfigurationUpdatedWithRevision")
+
+	if err != nil {
+		t.Fatalf("Error fetching Revision %v", err)
+	}
+
+	revision, err := clients.ServingClient.Revisions.Get(revName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching Revision %s: %v", revName, err)
+	}
+
+	if val, ok := revision.Labels["serving.knative.dev/configuration"]; ok {
+		if val != names.Config {
+			t.Fatalf("Expect confguration name in revision label %q but got %q ", names.Config, val)
+		}
+	} else {
+		t.Fatalf("Failed to get configuration name from Revision label")
+	}
+	if val, ok := revision.Labels["serving.knative.dev/service"]; ok {
+		if val != names.Service {
+			t.Fatalf("Expect Service name in revision label %q but got %q ", names.Service, val)
+		}
+	} else {
+		t.Fatalf("Failed to get Service name from Revision label")
+	}
```

Comment:
What I've done before to test this sort of thing end-to-end is to have a tiny container which is a webserver which just dumps it's environment. You can then deploy it to a known location and fetch the environment page to check to see if your labels are present.

(I'm okay if you want to leave this as a TODO on this code.)

Reply:
will add a new e2e test